requirements-dev.txt

```diff
@@ -7,13 +7,13 @@ stestr>=2.0.0
 ddt>=1.2.0,!=1.4.0,!=1.4.3
 reno>=3.4.0
 Sphinx>=5.0
-sphinx-design>=0.2.0
+sphinx-design~=0.4.1
```

Comment:
What is the rationale on this. I get it with the sphinx theme but its less than clear to me doing it with this.

Reply:
In general, I think it's helpful to pin docs because it improves code stability. For example, it reduces the risk of CI breaking overnight. But I can revert this to >=0.4.1 if you prefer.

Imo, the ideal is using loose requirements in the requirements.txt file, but then using a lockfile to pin every transitive dependency to an exact version. That way, you only make changes to deps when you intentionally want to. It substantially reduces the risk of things breaking overnight. But you also get the newest versions of deps when you regenerate the lockfile. If you're interested, I'm happy to set this up for you by using `pip-compile`, which is a popular tool to generate ~lockfiles.

Reply:
>=0.2.0 also works - or it least it always has.

We generally get more breakage from lint or mypy changing than anything else. If its pinned its likely to stay that way. Yes it sometimes CI breaks but at least it makes us keep up to date with the latest - or we temporarily pin it while things are changed if its more involved.